templates/cdap-deployment.yaml

```diff
@@ -97,7 +97,17 @@ spec:
           env:
           {{range $e := $c.Env }}
           - name: "{{$e.Name}}"
-            value: "{{$e.Value}}"
+            {{if $e.Value}}
```

Comment:
Like the other comment regarding moving the container to go code, we should do the same for environment variables if we want to support `ValueFrom`. This will significantly simplify the generation logic since we can just directly add the container/envvar to the YAML instead of relying on templating.

Additionally, can we also add this to cdap-sts.yaml to keep environment variables consistent across all containers? Thanks!

Reply:
Thanks for your review @dli357 . I have now moved the environment variables to go code. Could you please have a further look and let me know? Thank you